packages/block-library/src/navigation-menu/index.php

```diff
@@ -56,18 +56,25 @@ function build_navigation_menu_html( $block ) {
  * @uses render_block_navigation_menu()
  */
 function register_block_core_navigation_menu() {
+	$block_content = file_get_contents ( dirname( __FILE__ ) . '/../../../packages/block-library/src/navigation-menu/block.json' );
```

Comment:
It's a very neat idea 👍 

Gutenberg copies this PHP file to the build folder. I guess you should change the build process, to copy the JSON file as well and locate it in the same folder.

```suggestion
	$block_content = file_get_contents ( dirname( __FILE__ ) . '/block.json' );
```

This way it will work here and later in WordPress core as well.

Reply:
rookie feeling :-D Thanks!

Reply:
mmm, actually it doesn't seem to work. The path there is 
`/var/www/src/wp-content/plugins/gutenberg/build/block-library/blocks/block.json`
getting the following warning:

`
Warning: file_get_contents(/var/www/src/wp-content/plugins/gutenberg/build/block-library/blocks/block.json): failed to open stream: No such file or directory in /var/www/src/wp-content/plugins/gutenberg/build/block-library/blocks/navigation-menu.php
`

Reply:
We don't copy `block.json` files at the moment. You will have to change it to make it work. 

This is how PHP files are copied and transformed:
https://github.com/WordPress/gutenberg/blob/d8bb7529bb6bd40eb633e742a6423595367024df/webpack.config.js#L117-L121

It seems like the output path for PHP files doesn't take into account that there might be other files necessary. You will have to tweak it somehow.

In addition, JSON files aren't whitelisted for the plugin assets, so this is another place where changes are necessary:

https://github.com/WordPress/gutenberg/blob/master/bin/build-plugin-zip.sh#L112

Reply:
Maybe we should have a standard way of doing this for all blocks, instead of using file_get_contents and hardcoded paths, at least a constant being defined for each block or a param to the register function ...

Reply:
> It would be also a good idea to make it possible to pass the path of the block.json file as a param when registering the block so you don't have to repeat the same logic for all other server-side registered blocks.

> Maybe we should have a standard way of doing this for all blocks, instead of using file_get_contents and hardcoded paths, at least a constant being defined for each block or a param to the register function ...

Good ideas, but it needs to be handled in WordPress core, in the class `class-wp-block-type-registry.php`. Am I right? If so I'll happy to work on this but it will take more time than I considered initially with this PR.

However, I think it worths apply this change for this component and use it as a probe of the functionality.

Reply:
You can always start with a version of the function prefixed with `gutenberg_` before jumping into changes to the WordPress core. Once we agree on something in this PR, we can open a follow up which will enable it in core.